Remove ERGOM, TOPAZ, and miniBLING modules

yichengt900 opened the pull request:
As @nikizadehgfdl suggested in #21, the pull request (PR) removes the `ERGOM`, `TOPAZ`, and `miniBLING` modules, as well as associated variables and calls, from `generic_tracer.F90`.

charliestock commented:
Following discussion with Yi-Cheng, this should be OK as long as we have other module options as a template (e.g., BLING, CFCs), and then lets rename the .F90, .F90.archive for TOPAZ, minibling and ERGOM, and then add a README.  This should prevent compiling, and we can remove more completely if these modules continue to be unused.

yichengt900 committed:
rmove ERGOM, TOPAZ, and miniBLING

yichengt900 commented:
So instead of removing these modules directly, we rename them as `generic_*.F90.archive`, and we commented out all the related calls in `generic_tracer.F90` for now.

yichengt900 committed:
remove undefined namelists

yichengt900 committed:
small fix

andrew-c-ross commented on generic_tracers/generic_tracer.F90:
```diff
@@ -262,8 +262,8 @@ subroutine generic_tracer_init(isc,iec,jsc,jec,isd,ied,jsd,jed,nk,ntau,axes,grid
     !JGJ 2013/05/31  merged COBALT into siena_201303
-    if(do_generic_abiotic .or. do_generic_age .or. do_generic_argon .or. do_generic_CFC .or. do_generic_SF6 .or. do_generic_TOPAZ &
-       .or. do_generic_ERGOM .or. do_generic_BLING .or. do_generic_miniBLING .or. do_generic_COBALT .or. do_generic_blres) then
+    if(do_generic_abiotic .or. do_generic_age .or. do_generic_argon .or. do_generic_CFC .or. do_generic_SF6  &
+       .or. do_generic_BLING .or. do_generic_COBALT .or. do_generic_blres) then
```
Here the parts for TOPAZ, ERGOM, and miniBLING were deleted rather than commented out, so undoing this PR would not be as simple as un-commenting. To make it possible to revert while also simplifying the code, can we replace this whole statement with `if(do_generic_tracer) then`? The initialization currently gets called for every tracer that is used, and it seems like any other generic tracer that would be added would also need to initialize.

andrew-c-ross commented on generic_tracers/generic_tracer.F90:
```diff
@@ -645,8 +645,8 @@ subroutine generic_tracer_vertdiff_G(h_old, ea, eb, dt, kg_m2_to_H, m_to_H, tau)
     !JGJ 2013/05/31  merged COBALT into siena_201303
-    if(do_generic_abiotic .or. do_generic_age .or. do_generic_argon .or. do_generic_CFC .or. do_generic_SF6 .or. do_generic_TOPAZ &
-       .or. do_generic_ERGOM .or. do_generic_BLING .or. do_generic_miniBLING .or. do_generic_COBALT .or. do_generic_blres) then
+    if(do_generic_abiotic .or. do_generic_age .or. do_generic_argon .or. do_generic_CFC .or. do_generic_SF6  &
+       .or. do_generic_BLING .or. do_generic_COBALT .or. do_generic_blres) then
```
Also replace this with `if(do_generic_tracer) then`?

andrew-c-ross commented on generic_tracers/generic_tracer.F90:
```diff
@@ -687,8 +687,8 @@ subroutine generic_tracer_vertdiff_M(dh, dhw, diff_cbt, dt, Rho_0,tau)
     !JGJ 2013/05/31  merged COBALT into siena_201303
-    if(do_generic_age .or. do_generic_argon .or. do_generic_CFC .or. do_generic_TOPAZ .or. do_generic_ERGOM &
-       .or. do_generic_BLING .or. do_generic_miniBLING .or. do_generic_COBALT .or. do_generic_blres) then
+    if(do_generic_age .or. do_generic_argon .or. do_generic_CFC &
+       .or. do_generic_BLING .or. do_generic_COBALT .or. do_generic_blres) then
```
Also `if(do_generic_tracer) then` here? But, abiotic and SF6 are missing here. Can we actually delete generic_tracer_vertdiff_M (in a separate PR)?

andrew-c-ross replied:
#48

yichengt900 replied:
@andrew-c-ross, good point. Personally, I would love to remove unused modules or subroutines like `generic_tracer_vertdiff_M`, but I would like to hear from other people as well. Perhaps we can discuss this this afternoon.

andrew-c-ross commented on generic_tracers/generic_tracer.F90:
```diff
@@ -792,8 +792,8 @@ subroutine generic_tracer_coupler_set(IOB_struc, ST,SS,rho,ilb,jlb,tau, dzt, sos
     !
-    if(do_generic_abiotic .or. do_generic_age .or. do_generic_argon .or. do_generic_CFC .or. do_generic_SF6 .or. do_generic_TOPAZ &
-      .or. do_generic_ERGOM .or. do_generic_BLING .or. do_generic_miniBLING .or. do_generic_COBALT .or. do_generic_blres) &
+    if(do_generic_abiotic .or. do_generic_age .or. do_generic_argon .or. do_generic_CFC .or. do_generic_SF6  &
+      .or. do_generic_BLING .or. do_generic_COBALT .or. do_generic_blres) &
```
And here

yichengt900 committed:
simplify if statement

yichengt900 committed:
miss one if statement

yichengt900 committed:
Merge branch 'dev/cefi' into clean_old_modules

yichengt900 commented:
Hi @andrew-c-ross, I have addressed most of your comments except for the one regarding `generic_tracer_vertdiff_M` (which I will handle in a separate PR). Feel free to let me know if you have any further thoughts.
